quicklens/src/main/scala/com/softwaremill/quicklens/package.scala

```diff
@@ -115,12 +115,12 @@ package object quicklens {
     def each(fa: F[K, T])(f: T => T): F[K, T]
   }
 
-  implicit def mapQuicklensFunctor[K, T] = new QuicklensMapAtFunctor[Map, K, T] {
-    override def at(fa: Map[K, T], key: K)(f: T => T) = {
-      fa.updated(key, f(fa(key)))
+  implicit def mapQuicklensFunctor[M[KT, TT] <: Map[KT, TT], K, T] = new QuicklensMapAtFunctor[M, K, T] {
+    override def at(fa: M[K, T], key: K)(f: T => T) = {
+      fa.updated(key, f(fa(key))).asInstanceOf[M[K, T]]
```

Comment:
I would love to avoid the cast, but I did not find a way how. I have asked a question on the [code review](http://codereview.stackexchange.com/questions/119005/avoid-asinstanceof-in-a-type-class-implementation), perhaps someone will give some advice. CanBuildFrom sounds a bit like overkill to me - using that would mean rebuilding the whole map (and even if that seems worth it, I am not sure how would I do it). I would rather somehow explain to the compiler the type is right, because I know it is, but the `updated` function does not seem to carry this information through.